application/src/main/java/org/togetherjava/tjbot/features/help/MarkHelpThreadCloseInDBRoutine.java

```diff
@@ -59,14 +54,7 @@ private void updateTicketStatus(JDA jda) {
             .map(HelpThreadsRecord::getChannelId)
             .toList());
 
-
-        threadIdsToClose.forEach(id -> {
-            try {
```

Comment:
Please keep the try catch so that when one thread has a problem (whatever it might be), other threads arent stopped but still processed. every thread should get it's chance for closing.
the rest of the PR is perfect, thanks.

Reply:
@Zabuzard I don't understand, the only reason i earlier used try-catch was due to thread at times being nullable. Is it possible for something else to go wrong here that i can't anticipate?

Now we're handling the case where thread no longer exists in a much better way, imo exceptions should only reserved for something unexpected no?

Even if we do catch some exception is it something specific or just a general catch block?

Reply:
u should catch any exception happening here and log it. or collect it and rethrow it outside, after the `forEach`.

the reason is that if a thread cant be closed for any unknown reason, that should not stop the code trying to close the other threads. else they will be dangling.
like, imagine the code wants to cleanup 20 threads. thread 5 fails. what about the other 15 threads? they should also be attempted. without try-catch or similar, all 15 threads would be left opened bc the code crashed at thread 5.

so either try-catch-log or try-catch-collect+rethrow later